hypothesis-python/src/hypothesis/internal/observability.py

```diff
@@ -518,47 +422,22 @@ def _system_metadata() -> dict[str, Any]:
     }
 
 
-#: If ``False``, do not collect coverage information when observability is enabled.
-#:
-#: This is exposed both for performance (as coverage collection can be slow on
-#: Python 3.11 and earlier) and size (if you do not use coverage information,
-#: you may not want to store it in-memory).
-OBSERVABILITY_COLLECT_COVERAGE = (
-    "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_NOCOVER" not in os.environ
-)
-#: If ``True``, include the ``metadata.choice_nodes`` and ``metadata.spans`` keys
-#: in test case observations.
-#:
-#: ``False`` by default. ``metadata.choice_nodes`` and ``metadata.spans`` can be
-#: a substantial amount of data, and so must be opted-in to, even when
-#: observability is enabled.
-#:
-#: .. warning::
-#:
-#:     EXPERIMENTAL AND UNSTABLE. We are actively working towards a better
-#:     interface for this as of June 2025, and this attribute may disappear or
-#:     be renamed without notice.
-#:
-OBSERVABILITY_CHOICES = "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_CHOICES" in os.environ
-
-if OBSERVABILITY_COLLECT_COVERAGE is False and (
-    sys.version_info[:2] >= (3, 12)
-):  # pragma: no cover
-    warnings.warn(
-        "Coverage data collection should be quite fast in Python 3.12 or later "
-        "so there should be no need to turn coverage reporting off.",
-        HypothesisWarning,
-        stacklevel=2,
-    )
+envvar_observability: ObservabilityConfig | None = None
+
+# supported for backwards compat. These two were always marked experimental, so
+# they can be removed whenever. 6 months would be more than generous.
+if (
+    envvar_value := os.environ.get("HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY")
+) is not None:  # pragma: no cover
+    envvar_observability = ObservabilityConfig()
+
+if (
+    envvar_value := os.environ.get("HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY_CHOICES")
+) is not None:  # pragma: no cover
+    envvar_observability = ObservabilityConfig(choices=True)
 
 if (
-    "HYPOTHESIS_EXPERIMENTAL_OBSERVABILITY" in os.environ
-    or OBSERVABILITY_COLLECT_COVERAGE is False
-):  # pragma: no cover
-    add_observability_callback(_deliver_to_file, all_threads=True)
-
-    # Remove files more than a week old, to cap the size on disk
-    max_age = (date.today() - timedelta(days=8)).isoformat()
-    for f in storage_directory("observed", intent_to_write=False).glob("*.jsonl"):
-        if f.stem < max_age:  # pragma: no branch
-            f.unlink(missing_ok=True)
+    envvar_value := os.environ.get("HYPOTHESIS_OBSERVABILITY")
+) is not None:  # pragma: no cover
+    enabled = envvar_value in {"True", "true", "1"}
```

Comment:
I think here we want to explicitly error if it's set and _not_ in True/true/1/False/false/0; otherwise you can get the awkward situation where `=yes` is silently accepted, and then treated as false.